pilot/pkg/model/context.go

```diff
@@ -259,13 +259,14 @@ func ParseServiceNodeWithMetadata(s string, metadata map[string]string) (*Proxy,
 	}
 
 	out.ID = parts[2]
-	out.DNSDomain = parts[3]
+	out.DNSDomains = append([]string{parts[3]}, getProxyMetadataDNSDomains(out)...)
```

Comment:
this will result in duplicates. You need to follow same pattern as getProxyConfigNamesapce.. look at metadta first. If absent, take from parts[3]

Reply:
resolved with new implementation.